UpdateService/handler.ts

```diff
@@ -0,0 +1,199 @@
+
+import { Context } from "@azure/functions";
+import { left, right } from "fp-ts/lib/Either";
+import { identity } from "fp-ts/lib/function";
+import {
+  fromEither,
+  fromLeft,
+  taskEither,
+  TaskEither,
+  tryCatch
+} from "fp-ts/lib/TaskEither";
+import { ServiceModel } from "io-functions-commons/dist/src/models/service";
+import { ContextMiddleware } from "io-functions-commons/dist/src/utils/middlewares/context_middleware";
+import { RequiredParamMiddleware } from "io-functions-commons/dist/src/utils/middlewares/required_param";
+import { NonEmptyString } from "italia-ts-commons/lib/strings";
+import { Service } from "../generated/api-admin/Service";
+import { SubscriptionKeys } from "../generated/api-admin/SubscriptionKeys";
+import { ServicePayload } from "../generated/definitions/ServicePayload";
+import { ServiceWithSubscriptionKeys } from "../generated/definitions/ServiceWithSubscriptionKeys";
+import { APIClient } from "../utils/clients/admin";
+import {
+  ErrorResponses,
+  IResponseErrorUnauthorized,
+  ResponseErrorUnauthorized,
+  toErrorServerResponse
+} from "../utils/responses";
+
+type ResponseTypes =
+  | IResponseSuccessJson<ServiceWithSubscriptionKeys>
+  | IResponseErrorUnauthorized
+  | IResponseErrorForbiddenNotAuthorized
+  | IResponseErrorNotFound
+  | IResponseErrorTooManyRequests
+  | IResponseErrorInternal;
+
+/**
+ * Type of a GetUpdateService handler.
+ *
+ * GetUpdateService expects a service_id and a service payload as input
+ * and returns updated service with subscription keys
+ */
+type IGetCreateServiceHandler = (
+  context: Context,
+  auth: IAzureApiAuthorization,
+  clientIp: ClientIp,
+  attrs: IAzureUserAttributes,
+  serviceId: NonEmptyString,
+  servicePayload: ServicePayload
+) => Promise<ResponseTypes>;
+
+const getSubscriptionKeysTask = (
+  apiClient: ReturnType<APIClient>,
+  serviceId: NonEmptyString
+): TaskEither<ErrorResponses, SubscriptionKeys> =>
+  tryCatch(
+    () =>
+      apiClient.getSubscriptionKeys({
+        service_id: serviceId
+      }),
+    errs => ResponseErrorInternal(JSON.stringify(errs))
+  ).foldTaskEither(
+    err => fromLeft(err),
+    maybeResponse =>
+      maybeResponse.fold(
+        errs => fromLeft(ResponseErrorInternal(JSON.stringify(errs))),
+        responseType =>
+          responseType.status !== 200
+            ? fromLeft(toErrorServerResponse(responseType))
+            : taskEither.of(responseType.value)
+      )
+  );
+
+const updateServiceTask = (
+  apiClient: ReturnType<APIClient>,
+  servicePayload: ServicePayload,
+  serviceId: NonEmptyString,
+  subscriptionId: NonEmptyString
+): TaskEither<ErrorResponses, Service> =>
+  fromEither<IResponseErrorUnauthorized, {}>(
+    serviceId !== subscriptionId
+      ? left(
+          ResponseErrorUnauthorized(
+            "Unauthorized",
+            "You are not allowed to update this service"
+          )
+        )
+      : right({})
+  ).foldTaskEither(
```

Comment:
I'd keep this check in the caller

Reply:
Fixed in [98db5c7](https://github.com/pagopa/io-functions-services/pull/68/commits/98db5c77eefe2f2bbce99559596e00feed83aa0c)